### PR TITLE
Issue/4326 delete service button

### LINKED ISSUE #4326: [UX] Delete Service is not redirecting to catalog after deletion
# Issue
When you go to the Service Details page, and you delete a Service, you are not being redirected to the Service Catalog. Instead you get an error in your page with the suggestion to retry. Which serves no purpose since the service doesn't exist anymore.

# Solution
Remove the button in this view and move it as an action into the kebab menu on the catalog page.

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #4326 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design
Screenshot 2022-11-21 at 14 59 01

### CODE REVIEW

**Comment by LauraStordeur** [comment]
You got a cypres test failing, I suggest you delete the failing test for now, for the purpose to get the build succeeding, since we are going to rewrite them anyway.

File: changelogs/unreleased/4333-dependabot copy.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+change-type: patch
+description: 'Move delete service functionality to catalog view'
```
Comment (LauraStordeur) [inline changelogs/unreleased/4333-dependabot copy.yml:2]: add the section please :) minor-improvement should be fine

File: src/Slices/ServiceCatalog/UI/ServiceItem.tsx [hunk 2]
```diff
@@ -27,13 +29,33 @@ interface Props {
...
+  const handleModalToggle = () => {
+    setIsDeleteModalOpen(!isOpen);
+  };
+  const onSubmit = async () => {
+    handleModalToggle();
+    const result = await trigger();
+    if (Maybe.isSome(result)) {
+      setErrorMessage(result.value);
+    }
+  };
```
Comment (LauraStordeur) [inline src/Slices/ServiceCatalog/UI/ServiceItem.tsx:51]: can you keep new lines before return statements? It makes the code a bit more readable :)

File: changelogs/unreleased/4325-delete-service-button.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+change-type: patch
+description: 'Move delete service functionality to catalog view'
```
Comment (LauraStordeur) [inline changelogs/unreleased/4325-delete-service-button.yml:2]: Move delete service button to the Service Catalog overview, in the kebab menu options.

File: src/Slices/ServiceCatalog/UI/Page.test.tsx [hunk 4]
```diff
@@ -160,3 +160,26 @@ test("GIVEN ServiceCatalog WHEN new environment selected THEN new query is trigg
...
+
+test("GIVEN ServiceCatalog WHEN service is deleted THEN command is triggered", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+
+  await act(async () => {
+    await apiHelper.resolve(Either.right({ data: [Service.a] }));
+  });
+
+  await userEvent.click(screen.getByLabelText("Actions"));
```
Comment (LauraStordeur) [inline src/Slices/ServiceCatalog/UI/Page.test.tsx:172]: Actions should be from words dict

File: src/Slices/ServiceCatalog/UI/Page.test.tsx [hunk 5]
```diff
@@ -160,3 +160,26 @@ test("GIVEN ServiceCatalog WHEN new environment selected THEN new query is trigg
...
+  await userEvent.click(
+    screen.getByLabelText(Service.a.name + "-deleteButton")
+  );
+  await userEvent.click(screen.getByText("Yes"));
```
Comment (LauraStordeur) [inline src/Slices/ServiceCatalog/UI/Page.test.tsx:176]: words dict for Yes

File: src/Slices/ServiceCatalog/UI/Page.test.tsx (hunk 4 above)
Comment (matborowczyk) [reply src/Slices/ServiceCatalog/UI/Page.test.tsx:172]: aria-labe :D

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**